### PR TITLE
[FEAT]: add accordion to core folder

### PR DESCRIPTION
**Pull request by aleksandar-r**
Jira Ticket Link / Motivation
----

Accordion is being used on multiple places in a few of Ably's projects, so it should be implemented in AblyUI as a generic component, in order not to recreate it every time.

Summary of changes
----

Added Accordion component to AblyUI Core folder.

How do you manually test this?
----
In preview app find Accordion link under Components, click on it and try out the component for yourself.

Merge/Deploy Checklist
----

- [ ] Written automated tests for implemented features/fixed bugs
- [ ] Rebased and squashed commits
- [ ] Commits have clear descriptions of their changes
- [ ] Checked for any performance regressions

Frontend Checklist
----

- [ ] No frontend changes in this PR
- [ ] Added before/after screenshots for changes
- [ ] Tested on different platforms/browsers with [Browserstack](https://www.browserstack.com)
- [ ] Compared with the initial design / our [brand guidelines](https://www.figma.com/file/jTQgyIovrhGBOf4Zrvjvbk/Ably-Linear-Design?node-id=118%3A0)
- [ ] Checked the code for accessibility issues ([VoiceOver User Guide](https://support.apple.com/en-gb/guide/voiceover/welcome/mac))?

### CODE REVIEW

**Review by jamiehenson — COMMENTED** [review]
I know this is a WIP but when it's ready it would be good to see a Storybook story for this now that's available.

Also, side note, I've noticed this around `ably-ui` - do you know the purpose of having empty `.js` files along side a component? Is it a structural thing?

File: src/core/Accordion/component.tsx [hunk 1]
```diff
@@ -0,0 +1,143 @@
+import React, { useState, ReactNode, JSX } from "react";
+import Icon from "../Icon/component.jsx";
```
[the remark this answers is not in the document]
Comment (aleksandar-r) [reply src/core/Accordion/component.tsx:2]: Newest main is rebased, will change this

File: src/core/Accordion/component.tsx [hunk 2]
```diff
@@ -0,0 +1,143 @@
...
+
+export type AccordionData = {
```
[the remark this answers is not in the document]
Comment (aleksandar-r) [reply src/core/Accordion/component.tsx:4]: We need AccordionProps, so users know what props does it accept

**Review by jamiehenson — COMMENTED** [review]
Thanks for putting the story together 🏌️‍♂️ Good to see that it's keyboard navigable

I've a query about the animation - when closing an accordion slide while testing just now, the contents jump a bit, can you replicate this?
https://github.com/ably/ably-ui/assets/1105768/f5e4c00a-89e6-44cd-bb38-82c6e7f24c7e

**Review by jamiehenson — COMMENTED** [review]
Looks great ⛳ Last things before going in is to squash the commits (we only need 1 I think), and ideally to add a story to showcase the `autoClose` functionality as it's significant enough imo to not just be an optional control on the default story.

**Review by jamiehenson — APPROVED** [review]
🎩

**Pull request merged**